### PR TITLE
type(list): optimize triggerPaginationEvent type

### PR DESCRIPTION
**Pull request by thinkasany**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
PaginationConfig 中 有'onChange' | 'onShowSizeChange' 不需要用any
image
一共调用triggerPaginationEvent就以下两个地方
image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       optimize triggerPaginationEvent type    |
| 🇨🇳 Chinese |       优化triggerPaginationEvent type类型    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at cca5ed8</samp>

Refactored `triggerPaginationEvent` function in `components/list/index.tsx` to use a union type and a non-null assertion for better type safety. This was part of a TypeScript error fixing pull request.

### <samp>🤖 Generated by Copilot at cca5ed8</samp>

*  Refactor `triggerPaginationEvent` function to use union type for `eventName` parameter and assert non-undefined function ([link](https://github.com/ant-design/ant-design/pull/43905/files?diff=unified&w=0#diff-015bf7609c9e2715aacae53f1bbf8779c3f64b720462e1efc033b8d2711cd9e2L109-R116))

### CODE REVIEW

File: components/list/index.tsx [hunk 1]
```diff
@@ -106,13 +106,14 @@ function List<T>({
...
-    if (pagination && (pagination as any)[eventName]) {
-      (pagination as any)[eventName](page, pageSize);
-    }
-  };
+  const triggerPaginationEvent =
+    (eventName: 'onChange' | 'onShowSizeChange') => (page: number, pageSize: number) => {
+      setPaginationCurrent(page);
+      setPaginationSize(pageSize);
+      if (pagination && pagination[eventName]) {
+        pagination[eventName]!(page, pageSize);
```
Comment (li-jia-nan) [inline components/list/index.tsx:114]:
```suggestion
        pagination?.[eventName]?.(page, pageSize);
```

File: components/list/index.tsx [hunk 2]
```diff
@@ -106,13 +106,14 @@ function List<T>({
...
+      }
```
Comment (afc163) [inline components/list/index.tsx:115]:
```suggestion
      pagination?.[eventName]?.(page, pageSize);
```

File: components/list/index.tsx [hunk 3]
```diff
@@ -106,13 +106,12 @@ function List<T>({
...
-    setPaginationSize(pageSize);
-    if (pagination && (pagination as any)[eventName]) {
-      (pagination as any)[eventName](page, pageSize);
-    }
-  };
+  const triggerPaginationEvent =
+    (eventName: 'onChange' | 'onShowSizeChange') => (page: number, pageSize: number) => {
+      setPaginationCurrent(page);
+      setPaginationSize(pageSize);
+      (pagination as PaginationConfig)?.[eventName]?.(page, pageSize);
```
Comment (li-jia-nan) [inline components/list/index.tsx:113]: 这里还需要断言吗？

Comment (thinkasany) [reply components/list/index.tsx:113]: image

Comment (li-jia-nan) [reply components/list/index.tsx:113]: 明白了，if 判断去掉了，这里类型变宽了，是需要断言一下

Comment (li-jia-nan) [reply components/list/index.tsx:113]: @afc163 看了一下，这个 if 不能去掉，前后不等价了，这样可能会报错

Comment (afc163) [reply components/list/index.tsx:113]: 详述一下？为啥不等价了？

Comment (li-jia-nan) [reply components/list/index.tsx:113]: 可选链只能过滤 `undefined`，不能过滤 `false`，这里可能是 `false()`;

image

Comment (afc163) [reply components/list/index.tsx:113]: 图片

Comment (afc163) [inline components/list/index.tsx:113]:
```suggestion
      if (pagination) {
        (pagination as PaginationConfig)?.[eventName]?.(page, pageSize);
      }
```

File: components/list/index.tsx [hunk 4]
```diff
@@ -106,13 +106,14 @@ function List<T>({
...
-    if (pagination && (pagination as any)[eventName]) {
-      (pagination as any)[eventName](page, pageSize);
-    }
-  };
+  const triggerPaginationEvent =
+    (eventName: 'onChange' | 'onShowSizeChange') => (page: number, pageSize: number) => {
+      setPaginationCurrent(page);
+      setPaginationSize(pageSize);
+      if (pagination) {
+        (pagination as PaginationConfig)?.[eventName]?.(page, pageSize);
```
Comment (li-jia-nan) [inline components/list/index.tsx:114]:
```suggestion
        pagination?.[eventName]?.(page, pageSize);
```

Comment (li-jia-nan) [reply components/list/index.tsx:114]: 另外上面的 `ListProps<any>['pagination']` 可以顺便改成 `ListProps<T>['pagination']`

File: components/list/index.tsx [hunk 5]
```diff
@@ -106,13 +106,14 @@ function List<T>({
...
-    setPaginationSize(pageSize);
-    if (pagination && (pagination as any)[eventName]) {
-      (pagination as any)[eventName](page, pageSize);
-    }
-  };
+  const triggerPaginationEvent =
+    (eventName: 'onChange' | 'onShowSizeChange') => (page: number, pageSize: number) => {
+      setPaginationCurrent(page);
+      setPaginationSize(pageSize);
+      if (pagination) {
```
Comment (li-jia-nan) [inline components/list/index.tsx:113]:
```suggestion
      if (pagination && pagination[eventName]) {
```

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**